### PR TITLE
feat: add simple health HTTP endpoints

### LINKED ISSUE #961: Add readiness/liveness HTTP endpoints
This will allow to create probes for the Kubernetes DaemonSet to ensure rollouts are successful

Right now we hope it is running when this command returns:

https://github.com/parca-dev/parca-agent/blob/0e4ef533698b40c09d1295fb05dc1bd64ba04162/e2e/ci-e2e.sh#L73

### PR DESCRIPTION
**Pull request by maxbrunet**
Add simple health endpoints just to get started

Closes #961

### CODE REVIEW

File: cmd/parca-agent/main.go [hunk 1]
```diff
@@ -320,7 +320,7 @@ func run(logger log.Logger, reg *prometheus.Registry, flags flags) error {
 		),
 	}
 	mux.HandleFunc("/", func(w http.ResponseWriter, r *http.Request) {
-		if r.URL.Path == "/favicon.ico" {
+		if r.URL.Path == "/healthy" || r.URL.Path == "/ready" || r.URL.Path == "/favicon.ico" {
```
Comment (javierhonduco) [inline cmd/parca-agent/main.go:323]: nit: What do you think about`/status`? I think it is a more common health check path, no?

**Review by javierhonduco — APPROVED** [review]
LGTM!

File: cmd/parca-agent/main.go (hunk 1 above)
Comment (maxbrunet) [reply cmd/parca-agent/main.go:323]: `ready` and `healty` is a common naming for the Kubernetes-era, Prometheus, Thanos, and Kubernetes components use it for example. Later, we can imagine more in-depth checks under `/ready` and keep things light for continuous `/healthy` checks

Comment (javierhonduco) [reply cmd/parca-agent/main.go:323]: Sounds good! Guess I am not up to date on the Cloud Native ™️  lingo!

**Review by kakkoyun — APPROVED** [review]
It's good for a start. We have to do something more sophisticated in the long run. We have this prober pattern we use in Parca (or in other projects), where you can make sure if the app is actually healthy my changing the state of prober. So for us it should be only healthy if we have a running profiler, otherwise it should fail.

**Pull request merged**